cabal-lint/src/Cabal/Lint/Ord.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE LambdaCase #-}
+
+-- | This module defines wrappers with arbitrary 'Ord' instances around types
+-- for which @Cabal@ does not derive 'Ord'; we only use these to put these
+-- types in maps, sets, etc.
```

Comment:
Do you mean by this that the instances don't follow `Ord` type class laws?

Reply:
No; the `Ord` instances here are sound. But the `Cabal` devs have chosen for some reason to _not_ have them.

I assume that means these `Ord` instances are sound only if you're ignoring the data types' other semantics (ie it's a syntactical `Ord`).